File: rock_paper_scissors.py
import random

def get_computer_move ():
  computer_move = random.randint (1, 3)
  if computer_move == 1:
    return "rock"
  elif computer_move == 2:
    return "paper"
  else:
    return "scissors"

def check_player_win (computer, player):
  if computer == "rock" and player == "paper":
    return True
  elif computer == "paper" and player == "scissors":
    return True
  elif computer == "scissors" and player == "rock":
    return True
  else:
    return False

def check_player_loss (computer, player):
  if computer == "rock" and player == "scissors":
    return True
  elif computer == "paper" and player == "rock":
    return True
  elif computer == "scissors" and player == "paper":
    return True
  else:
    return False

def check_player_tie (computer, player):
  if computer == player:
    return True 
  else:
    return False 

print ("Rock, Paper, Scissors! \n")

player_move = input ("What is your move? ").lower()
computer_move = get_computer_move()

print ("\nYour move:", player_move)
print ("Computer move:", computer_move)

if check_player_win (computer_move, player_move) == True:
  print ("\nYou win!")
elif check_player_loss (computer_move, player_move) == True:
  print ("\nComputer wins!")
elif check_player_tie (computer_move, player_move) == True:
  print ("\nTie!")
else:
  print ("\nInvalid answer.")
  

File: C-Plus-Plus-Lessons/Chapter # 3/rock_paper_scissors.cpp
#include <iostream>

bool win (std::string player, std::string computer) {
  if (player == "rock" && computer == "scissors") {
    return true;
  } else if (player == "paper" && computer == "rock") {
    return true;
  } else if (player == "scissors" && computer == "paper") {
    return true;
  } else {
    return false;
  }
}

bool lose (std::string player, std::string computer) {
  if (computer == "rock" && player == "scissors") {
    return true;
  } else if (computer == "paper" && player == "rock") {
    return true;
  } else if (computer == "scissors" && player == "paper") {
    return true;
  } else {
    return false;
  }
}

bool tie (std::string player, std::string computer) {
  if (computer == player) {
    return true;
  }  else {
    return false;
  }
}

std::string get_computer () {
  srand(time (0));
  int n = rand() % 3;
  if (n == 0) {
    return "rock";
  } else if (n == 1) {
    return "paper";
  } else {
    return "scissors";
  }
}

bool valid (std::string move) {
  if (move == "paper" || move == "scissors" || move == "rock") {
    return true;
  } else {
    return false;
  }
}

int main() {
  
  std::cout << "\nWelcome to Rock, Paper, Scissors!\n";

  std::string move;
  std::cout << "\nYour move: ";
  std::cin >> move;

  if (valid (move)) {
    std::string computer = get_computer();
    std::cout << "Computer move: " << computer;
    if (win (move, computer)) {
      std::cout << "\nYou win!\n";
    } else if (tie (move, computer)) {
      std::cout << "\nYou tie!\n";
    } else {
      std::cout << "\nYou lost!\n";
    }
  } else {
    std::cout << "\nInvalid move.\n";
  }

}
